### ISSUE TITLE
Lack of trait implementation prevents freenet-email-app from building

### ISSUE THREAD

**Issue by KristijanZic** [freenet/freenet-core#846]
```sh
kristijanzic@mbp2 ~/Projects/Freenet/freenet-core/apps/freenet-email-app$ gmake build  

cargo install --force --path /Users/kristijanzic/Projects/Freenet/freenet-core/crates/core
cargo install --force --path /Users/kristijanzic/Projects/Freenet/freenet-core/crates/fdev
  Installing freenet v0.0.4 (/Users/kristijanzic/Projects/Freenet/freenet-core/crates/core)
    Updating crates.io index
    Finished release [optimized] target(s) in 6.27s
   Replacing /Users/kristijanzic/.cargo/bin/freenet
    Replaced package `freenet v0.0.4 (/Users/kristijanzic/Projects/Freenet/freenet-core/crates/core)` with `freenet v0.0.4 (/Users/kristijanzic/Projects/Freenet/freenet-core/crates/core)` (executable `freenet`)
  Installing fdev v0.0.4 (/Users/kristijanzic/Projects/Freenet/freenet-core/crates/fdev)
    Updating crates.io index
    Finished release [optimized] target(s) in 2.32s
   Replacing /Users/kristijanzic/.cargo/bin/fdev
    Replaced package `fdev v0.0.4 (/Users/kristijanzic/Projects/Freenet/freenet-core/crates/fdev)` with `fdev v0.0.4 (/Users/kristijanzic/Projects/Freenet/freenet-core/crates/fdev)` (executable `fdev`)
FILE="/Users/kristijanzic/Projects/Freenet/freenet-core/modules/identity-management/build/identity-manager-key.private.pem"
if [ ! -f $FILE ]; then
cd /Users/kristijanzic/Projects/Freenet/freenet-core/modules/identity-management && cargo run -- --path "/Users/kristijanzic/Projects/Freenet/freenet-core/modules/identity-management/build/"
fi
cd /Users/kristijanzic/Projects/Freenet/freenet-core/apps/freenet-email-app/contracts/inbox
fdev build --features contract
hash=$(bash -c "fdev inspect build/freenet/freenet_email_inbox code | grep 'code hash:' | cut -d' ' -f3")
mkdir -p /Users/kristijanzic/Projects/Freenet/freenet-core/apps/freenet-email-app/web/src/../build
echo -n $hash > /Users/kristijanzic/Projects/Freenet/freenet-core/apps/freenet-email-app/web/src/../build/inbox_code_hash
Compiling contract with rust
warning: use of deprecated associated function `chrono::DateTime::<Tz>::from_utc`: Use TimeZone::from_utc_datetime() or DateTime::from_naive_utc_and_offset instead
   --> /Users/kristijanzic/Projects/Freenet/freenet-core/modules/antiflood-tokens/interfaces/src/lib.rs:317:22
    |
317 |     DateTime::<Utc>::from_utc(naive, Utc)
    |                      ^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: `freenet-aft-interface` (lib) generated 1 warning
   Compiling freenet-email-inbox v0.0.1 (/Users/kristijanzic/Projects/Freenet/freenet-core/apps/freenet-email-app/contracts/inbox)
warning: unused variable: `pk`
   --> contracts/inbox/src/lib.rs:237:33
    |
237 | ...                   let pk = message
    |                           ^^ help: if this is intentional, prefix it with an underscore: `_pk`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: `freenet-email-inbox` (lib) generated 1 warning (run `cargo fix --lib -p freenet-email-inbox` to apply 1 suggestion)
    Finished release [optimized] target(s) in 2.02s
Contract compiled
cd /Users/kristijanzic/Projects/Freenet/freenet-core/apps/freenet-email-app/web
dx build --target index.html --release --platform web --features use-node
sed -i 's@/./assets@./assets@g' dist/index.html
fdev build
[INFO] 🚅 Running build command...
Error: 🚫 Building project failed: error[E0277]: the trait bound `freenet_stdlib::prelude::Parameters<'_>: From<identity_management::IdentityParams>` is not satisfied
   --> web/src/api.rs:360:43
    |
360 |         let params = Parameters::try_from(params)?;
    |                      -------------------- ^^^^^^ the trait `From<identity_management::IdentityParams>` is not implemented for `freenet_stdlib::prelude::Parameters<'_>`
    |                      |
    |                      required by a bound introduced by this call
    |
    = help: the following other types implement trait `From<T>`:
              <freenet_stdlib::prelude::Parameters<'a> as From<std::vec::Vec<u8>>>
              <freenet_stdlib::prelude::Parameters<'a> as From<&'a [u8]>>
    = note: required for `identity_management::IdentityParams` to implement `Into<freenet_stdlib::prelude::Parameters<'_>>`
    = note: required for `freenet_stdlib::prelude::Parameters<'_>` to implement `TryFrom<identity_management::IdentityParams>`


sed: 1: "dist/index.html": extra characters at the end of d command
Compiling contract with rust
error: Broken pipe (os error 32)
    Blocking waiting for file lock on build directory
error: could not compile `freenet-email-ui` (lib) due to 6 previous errors; 10 warnings emitted
    Finished release [optimized] target(s) in 0.24s
Contract compiled
Bundling webapp contract state
Error: unknown directory: "dist"
gmake: *** [Makefile:52: build-webapp] Error 1
```

neofetch output:
```sh
OS: macOS 13.4 22F66 arm64
Host: MacBookPro18,3
Kernel: 22.5.0
Uptime: 11 days, 7 hours, 2 mins
Packages: 245 (brew)
Shell: zsh 5.9
Resolution: 1512x982
DE: Aqua
WM: Rectangle
Terminal: /dev/ttys026
CPU: Apple M1 Pro
GPU: Apple M1 Pro
Memory: 3157MiB / 16384MiB
```

**Comment by iduartgomez** [comment]
Try doing a `cargo clean` and build again and see if that fixes it. This is a  problem with cargo I think where there is a conflict between using the local path dependency when building core and the published version because they are theoretically the same version.

**Comment by iduartgomez** [comment]
Everything should be fixed with the release of #853 

Reopen if still having problems (try from a  clean environment first though).
